src/shared/redux/slices/publicationsSlice.ts

```diff
@@ -1,57 +1,113 @@
-import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { createSlice, current, PayloadAction } from '@reduxjs/toolkit';
 import { RootState } from '../rootReducer';
 
-type CollaboratorElement = {
-  github_username: string;
+type Collaborator = {
+  id: string;
+  githubUsername: string;
   role: string;
 };
 
-type PublicationListElement = {
-  project_name: string;
-  collaborators: [CollaboratorElement];
-  pm_preference: string;
-  description: string;
+type Publication = {
+  id: string;
   dirPath: string;
+  publicationName: string;
+  description: string;
+  collaborators: Collaborator[];
+  packageManager: string;
+  useSass: string;
+  useTypescript: string;
 };
 
-type PublicationList = {
-  list: [PublicationListElement];
-};
+// type created to exclude collaborator type to avoid an issue with
+// index access writes on union of keys, which would complicate the
+// code significantly more
+type PublicationPrimitives = Omit<Publication, 'collaborators'>;
 
-const initialState: PublicationList = {
-  list: [
-    {
-      project_name: '',
-      collaborators: [{ github_username: '', role: '' }],
-      pm_preference: '',
-      description: '',
-      dirPath: '',
-    },
-  ],
+type PublicationModification = {
+  id: string;
+  field: keyof PublicationPrimitives;
+  value: string;
 };
 
+interface CollaboratorListModification<T> {
+  id: string;
+  value: T;
+}
+
+const initialState: Publication[] = [];
+
 const publicationsSlice = createSlice({
   name: 'publications',
   initialState,
   reducers: {
+    setPublicationsList: (
+      state: Publication[],
+      action: PayloadAction<Publication[]>
+    ) => action.payload,
     addPublication: (
-      state: PublicationList,
-      action: PayloadAction<PublicationListElement>
+      state: Publication[],
+      action: PayloadAction<Publication>
+    ) => {
+      state.push(action.payload);
+      console.log(current(state));
```

Comment:
I do not think we need these logs at all. The project comes in with [Redux dev tools](https://github.com/zalmoxisus/redux-devtools-extension) built into the development environment as set up in #86. The dev tools give a lot more control and information than the output of a log into console. 

#### Small preview

![image](https://user-images.githubusercontent.com/34404855/139946454-b414f99e-35eb-4611-8b90-50e5edb4ab6c.png)